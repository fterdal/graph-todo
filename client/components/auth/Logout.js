import React, { Component } from 'react';
import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';
import { meQuery } from '../SideBar';

export class Logout extends Component {

  _logout = async () => {
    await this.props.logoutMutation({
      refetchQueries: [
        { query: meQuery },
      ],
    })
  }

  render() {
    const { _logout } = this;
    return (
      <a
        className="button auth-button"
        onClick={() => _logout()}>
        Logout
      </a>
    )
  }
}

const logoutMutation = gql`
  mutation {
    logout {
      id
    }
  }
`;

export default compose(
  graphql(logoutMutation, {
    props: ({ mutate }) => ({ logoutMutation: mutate }),
  }),
)(Logout)
