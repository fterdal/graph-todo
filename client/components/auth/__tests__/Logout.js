/* eslint-env jest */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Logout } from '../Logout';

test('Logout renders', () => {
  const login = shallow(
    <Logout />
  );
  expect(login).toMatchSnapshot();
});

test('User can click to log out', () => {
  const logoutMutationMock = jest.fn();
  const logout = shallow(
    <Logout logoutMutation={logoutMutationMock} />
  );
  logout.find('.auth-button').simulate('click');
  expect(logoutMutationMock).toHaveBeenCalled();
});
