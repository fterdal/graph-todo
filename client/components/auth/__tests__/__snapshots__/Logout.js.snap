// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders 1`] = `
<a
  className="button auth-button"
  onClick={[Function]}
>
  Logout
</a>
`;
